cocaine/detail/secadaptor.py

```diff
@@ -125,18 +166,27 @@ def wrapper(*args, **kwargs):
         return wrapper
 
 
-class SecureServiceFabric(object):
+class SecureServiceFactory(object):
 
     @staticmethod
-    def make_secure_adaptor(service, mod, client_id, client_secret, tok_update_sec=None):
+    def make_secure_adaptor(service, endpoints=None, tok_update_sec=None, **secure_cfg):
         """
         :param service: Service to wrap in.
-        :param mod: Name (type) of token refresh backend.
-        :param client_id: Client identifier.
-        :param client_secret: Client secret.
+        :param endpoints: Token service endpoints. If value is None, endpoints to service will be used.
         :param tok_update_sec: Token update interval in seconds.
+        :param secure_cfg: Configuration for security provider.
         """
-        if mod == 'TVM':
-            return SecureServiceAdaptor(service, TVM(client_id, client_secret), tok_update_sec)
 
-        return SecureServiceAdaptor(service, Promiscuous(), tok_update_sec)
+        mod = secure_cfg.get('mod', '').upper()
+        token_service_endpoints = endpoints or service.endpoints
+        credentials = Credentials(secure_cfg['client_id'], secure_cfg['client_secret'])
+
+        provider = Promiscuous()
+        if mod == TVM.TYPE:
+            token_service_name = secure_cfg.get('service_name', 'tvm')
```

Comment:
Not important, but default service name could be part of class contract: _TVM.NAME_.